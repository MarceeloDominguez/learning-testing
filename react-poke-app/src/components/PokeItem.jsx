export default function PokeItem({ poke }) {
  return (
    <div>
      <h1>{poke.name}</h1>
      <span>#{poke.id}</span>
      <img src={poke.picture} alt={poke.name} />
    </div>
  );
}
