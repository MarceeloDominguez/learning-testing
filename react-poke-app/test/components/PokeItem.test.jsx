import { render, screen } from "@testing-library/react";
import PokeItem from "../../src/components/PokeItem";

describe("Testing <PokeItem />", () => {
  const poke = {
    name: "Pikachu",
    id: "1",
    picture:
      "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png",
  };

  test("should match with the snatshop", () => {
    const { container } = render(<PokeItem poke={poke} />);
    expect(container).toMatchSnapshot();
  });

  test("should show the image with the indicated URL and ALT", () => {
    render(<PokeItem poke={poke} />);

    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(poke.picture);
    expect(alt).toBe(alt);
  });

  test("should show the title in the component", () => {
    render(<PokeItem poke={poke} />);
    expect(screen.getByText(poke.name)).toBeTruthy();
  });
});
