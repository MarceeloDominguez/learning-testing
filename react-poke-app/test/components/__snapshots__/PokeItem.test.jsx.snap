// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <PokeItem /> should match with the snatshop 1`] = `
<div>
  <div>
    <h1>
      Pikachu
    </h1>
    <span>
      #
      1
    </span>
    <img
      alt="Pikachu"
      src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png"
    />
  </div>
</div>
`;
